sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -482,13 +482,105 @@ private[parquet] class ParquetRowConverter(
      */
     // scalastyle:on
     private def isElementType(
-        parquetRepeatedType: Type, catalystElementType: DataType, parentName: String): Boolean = {
+        parquetRepeatedType: Type, catalystElementType: DataType, parent: GroupType): Boolean = {
       (parquetRepeatedType, catalystElementType) match {
-        case (t: PrimitiveType, _) => true
-        case (t: GroupType, _) if t.getFieldCount > 1 => true
-        case (t: GroupType, _) if t.getFieldCount == 1 && t.getName == "array" => true
-        case (t: GroupType, _) if t.getFieldCount == 1 && t.getName == parentName + "_tuple" => true
-        case (t: GroupType, StructType(Array(f))) if f.name == t.getFieldName(0) => true
+        case (t: PrimitiveType, _) =>
+          // For legacy 2-level list types with primitive element type, e.g.:
+          //
+          //    // List<Integer> (nullable list, non-null elements)
+          //    optional group my_list (LIST) {           <-- parent
+          //      repeated int32 element;                 <-- repeatedType
+          //    }
+          true
+
+        case (t: GroupType, _) if t.getFieldCount > 1 =>
+          // For legacy 2-level list types whose element type is a group type with 2 or more fields,
+          // e.g.:
+          //
+          //    // List<Tuple<String, Integer>> (nullable list, non-null elements)
+          //    optional group my_list (LIST) {           <-- parent
+          //      repeated group element {                <-- repeatedType
+          //        required binary str (UTF8);
+          //        required int32 num;
```

Comment:
Also mention that `t.getFieldCount` is 2 (including `str` and `num`) for this case?